core/domain/summary_services.py

```diff
@@ -63,6 +63,16 @@ def get_displayable_exp_summary_dicts_matching_ids(exploration_ids):
     displayable_exp_summaries = []
     exploration_summaries = (
         exp_services.get_exploration_summaries_matching_ids(exploration_ids))
+    displayable_exp_summaries = get_displayable_exp_summary_dicts(
+        exploration_summaries, exploration_ids)
+
+    return displayable_exp_summaries
+
+def get_displayable_exp_summary_dicts(exploration_summaries, exploration_ids):
```

Comment:
Note: it does add a bit of extra computation time, but this is minimal, and in general it's better to avoid possible skew in the arguments (e.g. we want to avoid cases where exploration_summaries and exploration_ids don't match each other, which is theoretically possible if they're provided as independent args).